internal/scaffold/crd_test.go

```diff
@@ -122,7 +123,54 @@ spec:
     storage: true
 `
 
-func TestCRDNonGoProject(t *testing.T) {
+func TestCRDNonGoProjectDefault(t *testing.T) {
+	s, buf := setupScaffoldAndWriter()
+	s.Fs = afero.NewMemMapFs()
+
+	r, err := NewResource(appApiVersion, appKind)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	crd := &CRD{Resource: r}
+	cfg := &input.Config{}
+	if err = s.Execute(cfg, crd); err != nil {
+		t.Fatalf("Failed to execute the scaffold: (%v)", err)
+	}
+
+	if crdNonGoDefaultExp != buf.String() {
+		diffs := diffutil.Diff(crdNonGoDefaultExp, buf.String())
+		t.Fatalf("Expected vs actual differs.\n%v", diffs)
+	}
+}
+
+// crdNonGoDefaultExp is the default non-go CRD. Non-go projects don't have the
+// luxury of kubebuilder annotations.
+const crdNonGoDefaultExp = `apiVersion: apiextensions.k8s.io/v1beta1
+kind: CustomResourceDefinition
+metadata:
+  name: appservices.app.example.com
+spec:
+  group: app.example.com
+  names:
+    kind: AppService
+    listKind: AppServiceList
+    plural: appservices
+    singular: appservice
+  scope: Namespaced
+  subresources:
+    status: {}
+  validation:
+    openAPIV3Schema:
+      type: object
+      x-kubernetes-preserve-unknown-fields: true
```

Comment:
I understand that if we do not replace the `apiextensions.k8s.io/v1beta1` for `apiextensions.k8s.io/v1` the `x-kubernetes-preserve-unknown-fields` is not required at all.

Reply:
Correct. It is optional in `v1beta1` and required in `v1`. The idea is to continue using `v1beta1` since pre-1.16 clusters do not support `v1`, but to begin using structural schema-compliant CRDs to ease the transition to `v1`.